components/function-controller/internal/controllers/serverless/job.go

```diff
@@ -20,16 +20,21 @@ func (r *FunctionReconciler) isOnJobChange(instance *serverlessv1alpha1.Function
 	image := r.buildExternalImageAddress(instance)
 	buildStatus := r.getConditionStatus(instance.Status.Conditions, serverlessv1alpha1.ConditionBuildReady)
 
-	if len(deployments) == 1 && deployments[0].Spec.Template.Spec.Containers[0].Image == image && buildStatus != corev1.ConditionUnknown {
+	expectedJob := r.buildJob(instance, "")
+
+	if len(deployments) == 1 &&
+		deployments[0].Spec.Template.Spec.Containers[0].Image == image &&
+		buildStatus != corev1.ConditionUnknown &&
+		len(jobs) > 0 &&
+		r.mapsEqual(expectedJob.GetLabels(), jobs[0].GetLabels()) {
 		return false
 	}
 
-	expectedJob := r.buildJob(instance, "")
-
 	return len(jobs) != 1 ||
 		len(jobs[0].Spec.Template.Spec.Containers) != 1 ||
 		// Compare image argument
 		!r.equalJobs(jobs[0], expectedJob) ||
```

Comment:
We can check `r.mapsEqual(expectedJob.GetLabels(), jobs[0].GetLabels())` inside `r.equalJobs`.

Reply:
Hm... but it will trigger build when you change the labels. Labels don't affect generated docker image, so what is the point of building it again?

Reply:
Ah.. yes in second reconcile :)